env-var-docs/parser-package/src/env_var_docs/parser.py

```diff
@@ -0,0 +1,522 @@
+"""Provides functionality for parsing and visiting an environment variable
+documentation file.
+"""
+
+import dataclasses
+import typing
+import json
+
+UnparsedJSON = dict[str, typing.Any]
+
+
+@dataclasses.dataclass
+class Group:
+    group_description: str
+    members: dict[str, UnparsedJSON]
+
+
+@dataclasses.dataclass
+class RegexTest:
+    val: str
+    should_match: bool
+
+
+@dataclasses.dataclass
+class Variable:
+    description: str
+    type: str
+    required: bool
+    values: list[str] | None
+    regex: str | None
+    regex_tests: list[RegexTest] | None
+
+
+@dataclasses.dataclass
+class ParseError:
+    path: str
+    """The JSON path of the invalid object."""
+
+    msg: str
+    """"Why the object is invalid."""
+
+
+ParseErrors = list[ParseError]
+
+
+@dataclasses.dataclass
+class NodeParseError:
+    """As the environment variable documentation file is parsed, any invalid
+    structures are reported through ParseErrors. A ParseError means that
+    neither a Group nor a Variable was able to be parsed.
+    """
+
+    path: str
+    """The JSON path of the invalid object."""
+
+    group_errors: ParseErrors
+    """Errors from attempting to parse as a Group."""
+
+    variable_errors: ParseErrors
+    """Errors from attempting to parse as a Variable."""
+
+
+@dataclasses.dataclass
+class NodeInfo:
```

Comment:
**suggestion (non-blocking):** why not `Node` instead of `NodeInfo`?

Reply:
No reason.  Changed to Node.